Sources/SwiftSyntaxBuilder/Trivia+Extension.swift

```diff
@@ -0,0 +1,45 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+extension Trivia {
+  /// Creates a new `Trivia` and adds spacing if needed.
+  ///
+  /// If there is 3 or more, then we see if the first two is a newline and a space
+  /// If so, we need to replace add the correct amount of space after each newline
+  /// comming after to have the right indentation of the code.
```

Comment:
This comment was a little confusing to me. I’m not sure if my suggestion is better but I’ll give it a try:

````
Re-indents trivia that form a block comment to have consistent indentation.

If the trivia start with a newline (i.e. the first non-whitespace trivia is on a fresh line) and that newline starts with spacing, the same spacing is applied to all the following lines. For example.

```
  // first line
// second line
// third line
```
picks up the indentation from the first line and transforms the trivia to
```
  // first line
  // second line
  // third line
```
````